Shared/mods/deathmatch/logic/lua/CLuaFunctionParser.h

```diff
@@ -536,7 +536,11 @@ struct CLuaFunctionParserBase
         else if constexpr (std::is_same_v<T, CVector2D>)
         {
             if (lua_isnumber(L, index))
-                return {PopUnsafe<float>(L, index), PopUnsafe<float>(L, index)};
```

Comment:
Why would we need to get rid of list initialization?

Reply:
Better to explicitly return a class rather than have the compiler do some magic behind it that may brind unexpected consequences.

Reply:
How about `CVector2D{PopUnsafe<float>(L, index), PopUnsafe<float>(L, index)}`?

Reply:
it may or may not be specifed depending on the C++ version.
Better to define x,y,z beforehand

Reply:
I don't expect that C++ version will be downgraded some day. That means that CVector2D{} will work in the foreseeable future without any problems. Your approach is cumbersome and can lead to performance problems(depends on the compiler).

Reply:
There are no performance problems here. Either way it will still call the constructor. Doesnt matter if its `()` or `{}`. The performance difference is imperceptible

Reply:
Your current implementation unreasonably inflates the codebase and you should stick to the original list initialization.

Reply:
Original implementation is confusing and a small change breaks the entire codebase.
It absolutely does not inflate the codebase. Function calls that modify global objects (including themselves) should be cached beforehand.